akka-docs/rst/scala/code/docs/ddata/protobuf/TwoPhaseSetSerializer.scala

```diff
@@ -51,9 +51,11 @@ class TwoPhaseSetSerializer(val system: ExtendedActorSystem)
 
   def twoPhaseSetFromBinary(bytes: Array[Byte]): TwoPhaseSet = {
     val msg = TwoPhaseSetMessages.TwoPhaseSet.parseFrom(bytes)
-    TwoPhaseSet(
-      adds = GSet(msg.getAddsList.iterator.asScala.toSet),
-      removals = GSet(msg.getRemovalsList.iterator.asScala.toSet))
+    val addsSet = msg.getAddsList.iterator.asScala.toSet
+    val removalsSet = msg.getRemovalsList.iterator.asScala.toSet
+    val adds = addsSet.foldLeft(GSet.empty[String])((acc, el) => acc.add(el))
+    val removals = removalsSet.foldLeft(GSet.empty[String])((acc, el) => acc.add(el))
+    TwoPhaseSet(adds, removals)
```

Comment:
I think this is looking good. Perhaps

    // GSet will accumulate deltas when adding elements,
    // but those are not of interest in the result of the deserialization
    TwoPhaseSet(adds.resetDeltas, removals.resetDeltas)